Lib/subprocess.py

```diff
@@ -218,22 +218,39 @@ def __repr__(self):
         _PopenSelector = selectors.SelectSelector
 
 
-# This lists holds Popen instances for which the underlying process had not
-# exited at the time its __del__ method got called: those processes are wait()ed
-# for synchronously from _cleanup() when a new Popen object is created, to avoid
-# zombie processes.
-_active = []
-
-def _cleanup():
-    for inst in _active[:]:
-        res = inst._internal_poll(_deadstate=sys.maxsize)
-        if res is not None:
-            try:
-                _active.remove(inst)
-            except ValueError:
-                # This can happen if two threads create a new Popen instance.
-                # It's harmless that it was already removed, so ignore.
-                pass
+if _mswindows:
+    # On Windows we just need to close `Popen._handle` when we no longer need
+    # it, so that the kernel can free it. Keeping the handle open will actually
+    # create a zombie until the next `_cleanup()` call, which may be never if
+    # `Popen()` isn't called again. `CloseHandle` is called as a `__del__` in
+    # `class Handle` defined above, so there is no need to collect it as
+    # `_active` in `Popen.__del__` if it didn't finish yet.
```

Comment:
I'm now confused. This comment explains that __del__() must be called to avoid zombie process. But what is a process exit shortly, before __del__()? This comment basically explains that _active is needed, but then set it to None. I'm confused :-)